easybuild/framework/easyconfig/types.py

```diff
@@ -462,6 +462,8 @@ def to_checksums(checksums):
             for key, value in checksum.items():
                 validated_dict[key] = to_checksums(value)
             res.append(validated_dict)
+        else:
+            res.append(checksum)
```

Comment:
Wouldn't it be prudent to add None to the list of possible values in the "each list entry can be" list above (the comment that is)

Reply:
And also, doesn't this construct open up for having just an integer as the checksum value? which will probably cause problems...

Reply:
good point, fixed in a2786d66c

Reply:
what about my second comment? 
in other words, shouldn't that else: really check if it None instead of just being an else:?

Reply:
This is just the type auto-conversion stuff, not the checking of checksum values.
It takes care of changing `[['sha256', '<sha256_checksum>']]` to `[('sha256', '<sha256_checksum>')]` for example (which is partially done because of the `.yep` easyconfigs format).

Anything that is not a valid checksum value (like an integer) will still result in a failure when the checksum values are actually picked up in `easyblock.py`.